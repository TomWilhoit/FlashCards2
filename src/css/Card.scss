$font-title: "Roboto Condensed", sans-serif;
$font-copy: "Lato", sans-serif;
$btn-font: "Abel", sans-serif;

$btn-color: #0c3e9c;
$font-color: #fff;

.answer-btns {
  display: flex;
  justify-content: center;
  flex-direction: column;
  font-size: 0.8em;
  margin: auto;
}

.answer-btn {
  width: 100%;
  background: #0c3e9c;
  color: #fff;
  border-radius: 5px;
  font-family: $font-copy;
  font-size: 18px;
  padding: 1em;
  margin: 1em 0;
  box-shadow: 0px 1px 2px 2px black;
  border: none;
  font-weight: 100;
  cursor: pointer;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;

  &:hover {
    background: #000000;
  }
}

.Card {
  max-width: 650px;
  margin: auto;
  background: #3a3a3a;
  padding: 1em 2em;
  border-radius: 5px;
  box-shadow: -1px 3px 15px 4px black;
}

.question {
  font-family: $font-title;
  text-transform: uppercase;
  color: #fff;
  font-weight: 100;
}
