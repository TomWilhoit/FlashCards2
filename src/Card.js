import React, { Component } from "react";
import "./css/Card.css";

class Card extends Component {
  constructor(props) {
    super(props);
    this.state = {
      buttonOrder: 0
    };
  }

  buttonRandomize = () => {
    this.setState({
      buttonOrder: Math.floor(Math.random() * (3 - 1 + 1)) + 1
    });
  };

  correctGuess = () => {
    this.props.incrementQuestionIndex();
    this.props.shouldRepeatQuestions();
    this.buttonRandomize();
  };

  falseGuess = () => {
    this.props.incrementQuestionIndex();
    this.props.saveToStorage();
    this.props.shouldRepeatQuestions();
    this.buttonRandomize();
  };

  render() {
    if (this.state.buttonOrder === 1) {
      return (
        <div className="Card">
          <div>
            <h2 className="question">{this.props.questions.question}</h2>
            <button className="correct-answer-btn" onClick={this.correctGuess}>
              {this.props.questions.correctAnswer}
            </button>
            <button className="false-answer-btn1" onClick={this.falseGuess}>
              {this.props.questions.falseAnswer1}
            </button>
            <button className="false-answer-btn2" onClick={this.falseGuess}>
              {this.props.questions.falseAnswer2}
            </button>
          </div>
        </div>
      );
    } else if (this.state.buttonOrder === 2) {
      return (
        <div className="Card">
          <div>
            <h2 className="question">{this.props.questions.question}</h2>
            <button className="false-answer-btn1" onClick={this.falseGuess}>
              {this.props.questions.falseAnswer1}
            </button>
            <button className="correct-answer-btn" onClick={this.correctGuess}>
              {this.props.questions.correctAnswer}
            </button>
            <button className="false-answer-btn2" onClick={this.falseGuess}>
              {this.props.questions.falseAnswer2}
            </button>
          </div>
        </div>
      );
    } else {
      return (
        <div className="Card">
          <div>
            <h2 className="question">{this.props.questions.question}</h2>
            <button className="false-answer-btn1" onClick={this.falseGuess}>
              {this.props.questions.falseAnswer1}
            </button>
            <button className="false-answer-btn2" onClick={this.falseGuess}>
              {this.props.questions.falseAnswer2}
            </button>
            <button className="correct-answer-btn" onClick={this.correctGuess}>
              {this.props.questions.correctAnswer}
            </button>
          </div>
        </div>
      );
    }
  }
}

export default Card;
