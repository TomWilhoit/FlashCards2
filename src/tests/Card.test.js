import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';


describe('Card', () => {
  
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Card/>
      )
    });
    it('should match snapshot when all data is passed correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  })