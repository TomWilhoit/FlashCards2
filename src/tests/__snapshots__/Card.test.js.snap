// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot when all data is passed correctly 1`] = `
<div
  className="Card"
>
  <div>
    <h2
      className="question"
    >
      What is 'context' in Javascript?
    </h2>
    <div
      className="answer-btns"
    >
      <button
        className="answer-btn false-answer-btn1"
        onClick={[Function]}
      >
        Context is the exact same as scope
      </button>
      <button
        className="answer-btn false-answer-btn2"
        onClick={[Function]}
      >
        Function based reference to the function that 'owns' the currently executing code
      </button>
      <button
        className="answer-btn correct-answer-btn"
        onClick={[Function]}
      >
        The value of the this keyword, in reference to the object that owns the currently executing code
      </button>
    </div>
  </div>
</div>
`;
